### PR TITLE
Copyright header enforcement

### PR DESCRIPTION
**Pull request by pakrym**
Had a second and it was quick.

### CODE REVIEW

File: eng/CodeAnalysis.ruleset [hunk 1]
```diff
@@ -194,10 +194,10 @@
         <Rule Id="SA1630"  Action="None" />          <!-- Documentation text should contain whitespace -->
         <Rule Id="SA1631"  Action="None" />          <!-- Documentation should meet character percentage -->
         <Rule Id="SA1632"  Action="None" />          <!-- Documentation text should meet minimum character length -->
-        <Rule Id="SA1633"  Action="None" />          <!-- File should have header -->
-        <Rule Id="SA1634"  Action="None" />          <!-- File header should show copyright -->
-        <Rule Id="SA1635"  Action="None" />          <!-- File header should have copyright text -->
-        <Rule Id="SA1636"  Action="None" />          <!-- File header copyright text should match -->
+        <Rule Id="SA1633"  Action="Warning" />       <!-- File should have header -->
```
Comment (weshaggard) [inline eng/CodeAnalysis.ruleset]: Does this just enforce that one exists or does it enforce the value? If it enforces the value how does it know which one to require?

Comment (heaths) [reply eng/CodeAnalysis.ruleset]: I've been working with @chidozieononiwu with the files I used for the setup engine that enforced this. It's done through StyleCop and you can enforce non-XML headers. This change would enforce it, but the corresponding StyleCop.json changes (and how we put those as `AdditionalItems` items in each project) needs to be done as well.

File: eng/Directory.Build.Data.props [hunk 2]
```diff
@@ -53,6 +53,11 @@
     <CodeAnalysisRuleSet>$(RepoEngPath)\CodeAnalysis.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <!-- StyleCop configuration -->
+  <ItemGroup>
+    <AdditionalFiles Include="$(RepoEngPath)\stylecop.json" Link="stylecop.json" />
```
Comment (heaths) [inline eng/Directory.Build.Data.props]: IMO, I'd forget `Link` and set `Visible="false"`. These don't need to show up in ever project.

**Comment by chidozieononiwu** [comment]
Was in the process of sending the PR for this [here](https://github.com/Azure/azure-sdk-for-net/pull/7698). Did you already include stylecop.json. Looks like that is being excluded by gitignore

File: eng/Directory.Build.Data.props (hunk 2 above)
Comment (heaths) [reply eng/Directory.Build.Data.props]: I also don't see this file checked in yet. Perhaps dependent on @chidozieononiwu pending PR?

**Review by heaths — APPROVED** [review]

File: eng/CodeAnalysis.ruleset (hunk 1 above)
Comment (pakrym) [reply eng/CodeAnalysis.ruleset]: @chidozieononiwu was right, stylecop.json was ignored by .gitignore, it contains the required header value.

**Pull request merged**